ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/KtlintSuppressionRuleTest.kt

```diff
@@ -0,0 +1,798 @@
+                """
+                @SuppressWarnings("aaa", "ktlint:some-rule-set:some-rule-id", "zzz")
+                val foo = "foo"
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolation(2, 20, "Directive 'ktlint-disable' is deprecated. Replace with @Suppress annotation")
+                .isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given a ktlint-disable directive for which the target element is already annotated with both @Suppress and @SuppressWarnings then add the ktlint suppression to the @Suppress`() {
+            val code =
+                """
+                @Suppress("aaa", "zzz")
+                @SuppressWarnings("bbb", "yyy")
+                val foo = "foo" // ktlint-disable some-rule-set:some-rule-id
+                """.trimIndent()
+            val formattedCode =
+                """
+                @Suppress("aaa", "ktlint:some-rule-set:some-rule-id", "zzz")
+                @SuppressWarnings("bbb", "yyy")
+                val foo = "foo"
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolation(3, 20, "Directive 'ktlint-disable' is deprecated. Replace with @Suppress annotation")
+                .isFormattedAs(formattedCode)
+        }
+    }
+
+    @Nested
+    inner class `Given a top level block comment with a ktlint-disable directive` {
+        @Test
+        fun `Given a ktlint-disable directive without rule-id`() {
+            val code =
+                """
+                /* ktlint-disable */
+                val foo = "foo"
+                """.trimIndent()
+            val formattedCode =
+                """
+                @Suppress("ktlint")
+                val foo = "foo"
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolation(1, 4, "Directive 'ktlint-disable' is deprecated. Replace with @Suppress annotation")
+                .isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given a ktlint-disable directive with rule-id not prefixed with a rule set id`() {
+            val code =
+                """
+                /* ktlint-disable some-rule-id */
+                val foo = "foo"
+                """.trimIndent()
+            val formattedCode =
+                """
+                @Suppress("ktlint:standard:some-rule-id")
+                val foo = "foo"
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolation(1, 4, "Directive 'ktlint-disable' is deprecated. Replace with @Suppress annotation")
+                .isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given a ktlint-disable directive with rule-id prefixed with a rule set id`() {
+            val code =
+                """
+                /* ktlint-disable some-rule-set:some-rule-id */
```

Comment:
One TC can be added around with multiple `ktlint-disable`
```
/* ktlint-disable some-rule-set:some-rule-id-1 */
/* ktlint-disable some-rule-set:some-rule-id-2 */

/* ktlint-disable some-rule-set:some-rule-id-3 */
```

Reply:
This already existed in TC `Given multiple ktlint-disable directives which have to merged into an existing @file Suppress annotation`